application/server/main/tasks.py

```diff
@@ -112,81 +113,99 @@ def create_task_unpaywall(args):
         logger_console.debug(db_handler.fetch_all())
 
 
-def get_softcite_version(local_files):
-    softcite_files = [file for file in local_files if file.endswith('.software.json')]
-    if softcite_files:
-        with open(softcite_files[0], 'r') as f:
-            softcite_version = json.load(f)['version']
-    else:
-        softcite_version = "0"
+def get_softcite_version(softcite_file: str) -> str:
+    """Get the version of softcite used by reading from an output file"""
+    with open(softcite_file, 'r') as f:
+        softcite_version = json.load(f)['version']
     return softcite_version
 
 
-def get_grobid_version(local_files):
-    grobid_files = [file for file in local_files if file.endswith('.tei.xml')]
-    if grobid_files:
-        with open(CONFIG_PATH_GROBID, 'r') as f:
-            config = json.load(f)
-        url = f"http://{config['grobid_server']}:{config['grobid_port']}/api/version"
-        grobid_version = requests.get(url).text
-    else:
-        grobid_version = "0"
+def get_grobid_version() -> str:
+    """Get the version of grobid used by a request to the grobid route /api/version"""
+    with open(grobid_ns.config_path, 'r') as f:
+        config = json.load(f)
+    url = f"http://{config['grobid_server']}:{config['grobid_port']}/api/version"
+    grobid_version = requests.get(url).text
     return grobid_version
 
 
-def filter_publications(db_handler, spec_softcite_version, spec_grobid_version):
-    entries_publications_grobid = []
-    entries_publications_softcite = []
-    for record in db_handler.fetch_all():
-        uuid, doi = record[1], record[0]
-        softcite_version, grobid_version = record[3], record[4]
-        if softcite_version != spec_softcite_version:
-            entries_publications_softcite.append((doi, uuid))
-        if grobid_version != spec_grobid_version:
-            entries_publications_grobid.append((doi, uuid))
-    return entries_publications_softcite, entries_publications_grobid
+def get_publications_ids_to_process(db_records: List[ProcessedEntry], service_name: str, min_spec_version: str) -> List[Tuple[str]]:
+    """Return all the publications in the database that needs to be processed by a service (i.e. not just for the partition)"""
+    publications_ids_to_process = []
+    if service_name == "softcite":
+        version_key = 3
```

Comment:
Don't we have the column name in the record ? Because if we reorder the columns, everything will be confused. I would rather use something like `record[service_name]` if possible.

Reply:
record vient de db_handler.fetch_all() qui retourne des objets du domaine (ProcessedEntry) qui ont un attribut uuid.
Je fais le changement pour utiliser record.uuid à la place des indices là où c'est possible.